test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -315,14 +300,16 @@
            (is (true? (set-constants-and-verify pool-context 3)))
 
            ;; The flush should still not have completed at this point because
-           ;; the last instance from the old pool is still being borrowed
-           (is (not (realized? flush-complete)))
+           ;; the last instance from the old pool is still being borrowed.
+           ;; Wait for some arbitrary (but short) period of time to see that
+           ;; that the agent is still busy - presumably handling the flush.
+           (is (false? (await-for 50 pool-agent)))
```

Comment:
Yeah, it is racy - and I think you're right that it probably would be more likely to fail locally (and, therefore, help identify a problem with the flush) than in Travis with the relatively short timeout.  I don't think it's any more racy than what it's replacing, though.  I could make the wait longer to make it more likely that we'd catch a failure, but I'm also sensitive to drawing out the run time on this test too much since it seems likely that this test suite is going to get slower over time anyway.

I don't have any good ideas on how to make this more airtight and not sure if it's really worth it to spend more time thinking about it at this point.  What do you think?